internal/lsp/eval.go

```diff
@@ -88,33 +94,49 @@ func FindInput(file string, workspacePath string) io.Reader {
 	return nil
 }
 
-func (l *LanguageServer) EvalWorkspacePath(ctx context.Context, query string, input io.Reader) (EvalPathResult, error) {
+func (l *LanguageServer) EvalWorkspacePath(
+	ctx context.Context,
+	query string,
+	input io.Reader,
+) (EvalPathResult, error) {
 	resultQuery := "result := " + query
 
-	result, err := l.Eval(ctx, resultQuery, input)
+	hook := PrintHook{Output: make(map[int][]string)}
+
+	result, err := l.Eval(ctx, resultQuery, input, hook)
 	if err != nil {
 		return EvalPathResult{}, fmt.Errorf("failed evaluating query: %w", err)
 	}
 
 	if len(result) == 0 {
-		return EvalPathResult{IsUndefined: true}, nil
+		return EvalPathResult{IsUndefined: true, PrintOutput: hook.Output}, nil
 	}
 
 	res, ok := result[0].Bindings["result"]
 	if !ok {
 		return EvalPathResult{}, errors.New("expected result in bindings, didn't get it")
 	}
 
-	return EvalPathResult{Value: res}, nil
+	return EvalPathResult{Value: res, PrintOutput: hook.Output}, nil
```

Comment:
This is what's sent to the client right?

So it's up to the client to work out things like:

```
package play

import rego.v1

allow if {print(true)}
```

Reply:
Indeed!